src/components/ContentSidebar/ActivityFeed/comment/Mention.js

```diff
@@ -7,17 +7,58 @@ import * as React from 'react';
 
 type Props = {
     children?: React.Node,
-    id: number,
-    mentionTrigger?: any
+    id: string,
+    mentionTrigger?: any,
+    getUserProfileUrl?: (string) => Promise<string>
 };
 
-const Mention = ({ children, id, ...rest }: Props): React.Node => {
-    delete rest.mentionTrigger;
-    return (
-        <a {...rest} style={{ display: 'inline-block' }} href={`/profile/${id}`}>
-            {children}
-        </a>
-    );
+type State = {
+    profileUrl?: string
 };
 
+class Mention extends React.PureComponent<Props, State> {
+    state = {};
+
+    /**
+     * Success handler for getting avatar url
+     *
+     * @param {string} avatarUrl the user avatar url
+     */
+    getProfileUrlHandler = (profileUrl?: string) => {
```

Comment:
Is it possible that the get function fails? Maybe this could be called something like `getProfileUrlSuccess`, and you could have a corresponding `fail` that goes in a `catch` in the function below?